src/service/viewer-impl.js

```diff
@@ -208,6 +220,13 @@ export class Viewer {
     }
   }
 
+  onVisibilityChange_() {
+    if (this.isVisible()) {
+      this.whenVisibleResolve_();
+    }
+    this.visibilityObservable_.fire();
```

Comment:
There's one more place, I believe, where `this.visibilityObservable_.fire()` is called. Please search them all. Instead, this method should be called everywhere.